collect_app/src/main/java/org/odk/collect/android/utilities/Regex.java

```diff
@@ -0,0 +1,43 @@
+package org.odk.collect.android.utilities;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Created by simran on 4/8/2017.
+ */
+
+public class Regex {
+    public static final String GPS_LOCATION= "^-?[0-9]+\\.[0-9]+\\s-?[0-9]+\\.[0-9]+\\s-?[0-9]+\\"
+            + ".[0-9]+\\s[0-9]+\\.[0-9]+$";
+    public static final String VALID_GOOGLE_SHEETS_ID = "^[a-zA-Z0-9\\-]+$";
+    private static Pattern valid_google_sheets_id, gps_location;
+
+    /**
+     * Google sheets currently only allows a-zA-Z0-9 and dash
+     */
+    // Check to see if answer is a location, if so, get rid of accuracy
+    // and altitude
+    // try to match a fairly specific pattern to determine
+    // if it's a location
+    // [-]#.# [-]#.# #.# #.#
+
+
+    static {
+        valid_google_sheets_id = Pattern.compile(VALID_GOOGLE_SHEETS_ID);
```

Comment:
I like types and members to be defined and used in narrow scopes. So I wouldn’t move a regex (string, method, class, whatever) to a higher level just because it’s a regex.